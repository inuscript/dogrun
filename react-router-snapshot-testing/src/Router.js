import React from "react"
import { Switch, Route } from "react-router"

export default ({path = ""}) => {
  return <Switch>
    <Route path={`${path}/foo`}><div className="foo">foo</div></Route>
    <Route path={`${path}/baz`}><div className="baz">baz</div></Route>
    <Route path={`${path}/shallow`}><div><div className="shallowTarget">shallow</div></div></Route>
    <Route><div className="default">default</div></Route>
  </Switch>
}