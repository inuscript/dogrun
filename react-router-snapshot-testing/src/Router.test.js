import React from 'react';
import Router from "./Router"
import { StaticRouter, MemoryRouter } from "react-router-dom"
import { create } from 'react-test-renderer'
import { shallow, mount, render } from "enzyme"
const mockContext = {}

describe("snapshot", () => {
  it('default routing', () => {
    const tree = create(
      <StaticRouter context={mockContext}><Router /></StaticRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()  
  })
  
  it('/foo routing', () => {
    const tree = create(
      <StaticRouter context={mockContext} location="/foo"><Router /></StaticRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()  
  })

  it('/baz routing', () => {
    const tree = create(
      <StaticRouter context={mockContext} location="/baz"><Router /></StaticRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()  
  })

  it('memory routing', () => {
    const tree = create(
      <MemoryRouter context={mockContext}><Router /></MemoryRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()  
  })

})

describe("shallow", () => {

  it('/shallow shallow', () => {
    // shallowだとうまくfindが出来ない。renderかmountで可能
    const wrapper = render(
      <StaticRouter context={mockContext} location="/shallow"><Router /></StaticRouter>
    )
    // console.log(wrapper.())
    // console.log(wrapper.find(".shallowTarget").length)
    expect(wrapper.find(".shallowTarget").length).toBe(1)
  })

})