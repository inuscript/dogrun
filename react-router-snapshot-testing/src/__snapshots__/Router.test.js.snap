exports[`snapshot /baz routing 1`] = `
<div
  className="baz">
  baz
</div>
`;

exports[`snapshot /foo routing 1`] = `
<div
  className="foo">
  foo
</div>
`;

exports[`snapshot default routing 1`] = `
<div
  className="default">
  default
</div>
`;

exports[`snapshot memory routing 1`] = `
<div
  className="default">
  default
</div>
`;
